### PR TITLE
feat: Activity Log api

### PR DESCRIPTION
**Pull request by gurumarv**
### **How should this be manually tested?**

**Accessing the Endpoint:**

- Send a GET request to api/v1/organizations/:orgId/users/:userId/activity-logs.
- Ensure the Authorization header contains a valid bearer token.

**Testing Scenarios:**

- Valid Request: Use valid orgId and userId to retrieve activity logs.
- Invalid IDs: Test with invalid or non-existent orgId or userId.
- Unauthorized Access: Try accessing the endpoint without a valid authentication token.

**Request:**

**Endpoint:** api/v1/organizations/:orgId/users/:userId/activity-logs
**Method:** GET
**Headers:**
Authorization: Bearer <token>
**Parameters:**
orgId: The unique identifier of the organization (string).
userId: The unique identifier of the user (string).

**Successful Response (Status Code 200):**

Response Format:
```
{
  "message": "Activity logs retrieved successfully",
  "status_code": 200,
  "data": [
    {
      "activity": "User logged in",
      "timestamp": "Timestamp"
    },
    {
      "activity": "User updated profile",
      "timestamp": "Timestamp"
    }
  ]
}
```
**Error Response:**

**Not Found (Status Code 404):**
When the orgId or userId is invalid or non-existent.
```
{
  "status_code": 404,
  "message": "Organization or user not found",
  "error": "Not Found"
}
```
**Bad Request (Status Code 400):**
When an unauthenticated or unauthorized user tries to access the endpoint.
```
{
  "status_code": 400,
  "message": "Validation error message",
  "error": "Bad Request"
}
```

**Checklist:**

- [x] Define the API endpoint to retrieve user activity logs.
- [x]  Implement the logic to fetch activity logs from the database.
- [x]  Handle success and error responses appropriately.
- [x]  Validate orgId and userId parameters.
- [x]  Ensure proper authentication and authorization checks.

**Unit Tests:**

- [x] Successful response with valid orgId and userId.
- [x]  Error handling for invalid orgId or userId.
- [x]  Handle cases where orgId or userId is missing.
- [x]  Validate behavior when no activity logs are found.
- [x]  Confirm appropriate status codes are returned.
- [ ]

### CODE REVIEW

File: src/main/java/hng_java_boilerplate/activitylog/controller/ActivityLogController.java [hunk 1]
```diff
@@ -0,0 +1,63 @@
...
+        this.activityLogService = activityLogService;
+    }
+
+    @GetMapping("/{orgId}/users/{userId}/activity-logs")
+    public ResponseEntity<?> getActivityLogs(
+            @PathVariable("orgId") String orgId,
+            @PathVariable("userId") String userId,
+            Authentication authentication) {
+        try {
+            // Validate Strings
```
Comment (Fifetoyi) [inline src/main/java/hng_java_boilerplate/activitylog/controller/ActivityLogController.java:38]: remove comment

File: src/main/java/hng_java_boilerplate/activitylog/service/ActivityLogService.java [hunk 2]
```diff
@@ -0,0 +1,35 @@
...
+
+import java.time.Instant;
+import java.util.List;
+import java.util.UUID;
+
+@Service
+public class ActivityLogService {
+    private final ActivityLogRepository activityLogRepository;
+
+    @Autowired
```
Comment (Fifetoyi) [inline src/main/java/hng_java_boilerplate/activitylog/service/ActivityLogService.java:17]: @Autowired is no longer needed

File: src/main/java/hng_java_boilerplate/user/serviceImpl/UserServiceImpl.java [hunk 3]
```diff
@@ -81,22 +80,42 @@ public ResponseEntity<ApiResponse> loginUser(LoginDto loginDto) {
...
         }
 
         String token = jwtUtils.createJwt.apply(userDetails);
 
         UserResponse userResponse = getUserResponse(user);
         ResponseData data = new ResponseData(token, userResponse);
-        return new ResponseEntity<>(new ApiResponse(HttpStatus.OK.value(), "Login Successful!", data), HttpStatus.OK);
 
+        // Log activity
+        GetUserDto userDto = userRepository.findById(user.getId())
```
Comment (Fifetoyi) [inline src/main/java/hng_java_boilerplate/user/serviceImpl/UserServiceImpl.java:96]: can this be cleaned up to its own method?

File: src/main/java/hng_java_boilerplate/activitylog/controller/ActivityLogController.java [hunk 4]
```diff
@@ -0,0 +1,62 @@
...
+import java.util.stream.Collectors;
+
+@RestController
+@RequestMapping("/api/v1/organizations")
+@PreAuthorize("hasRole('ADMIN')")
+public class ActivityLogController {
+    private final ActivityLogService activityLogService;
+
+    @Autowired
+    public ActivityLogController(ActivityLogService activityLogService) {
```
Comment (lurldgbodex) [inline src/main/java/hng_java_boilerplate/activitylog/controller/ActivityLogController.java:28]: do you need autowired on constructor injection?

**Review by Fifetoyi — APPROVED** [review]
lgtm

**Pull request merged**